onnxscript/rewriter/pattern.py

```diff
@@ -891,64 +883,53 @@ def _apply_deltas(
-            # we'll try position i
-            assert i not in to_insert  # conflicts should avoid that case
-            to_insert.append((graph_or_function[i], inserted_nodes))
-        else:
-            deleted_nodes, inserted_nodes = delta
-            # Replace deleted nodes with inserted nodes.
-            # TODO: simplify this
-            last_deleted = deleted_nodes[-1]
-            last_inserted = inserted_nodes[-1]
-
-            for old_value, new_value in zip(last_deleted.outputs, last_inserted.outputs):
-                # Propagate relevant info from old value to new value
-                # TODO(Rama): Perhaps we should merge old and new types. As of now, new
-                # values don't have type information. Note that this could be a problem
-                # for semantics-altering rewrite-rules: we should allow users to override
-                # this for such rules.
-                new_value.type = old_value.type
-                new_value.shape = old_value.shape
-                new_value.const_value = old_value.const_value
-                new_value.name = old_value.name
-
-            # Reconnect the users of the deleted node to use the new outputs
-            _convenience.replace_all_uses_with(last_deleted.outputs, last_inserted.outputs)
-            # Update graph/function outputs if the node generates output
-            replacement_mapping = dict(zip(last_deleted.outputs, last_inserted.outputs))
-            for idx, graph_or_function_output in enumerate(graph_or_function.outputs):
-                if graph_or_function_output in replacement_mapping:
-                    graph_or_function.outputs[idx] = replacement_mapping[
-                        graph_or_function_output
-                    ]
-
-            # insert new nodes after the index node
-            graph_or_function.insert_after(last_deleted, inserted_nodes)
-            graph_or_function.remove(deleted_nodes, safe=True)
-
-    for replaced_node, inserted_nodes in to_insert:
-        graph_or_function.insert_after(replaced_node, inserted_nodes)
+
+    if isinstance(delta, tuple):
+        # multi-output strategy
+        n_matches, deleted_nodes, inserted_nodes = delta
+
+        # TODO(rama): Was "assert i not in to_insert"; seems wrong.
+        # What is this trying to check? Best effort correction below.
+        assert node not in inserted_nodes  # conflicts should avoid that case
+
+        graph_or_function.insert_after(node, inserted_nodes)
         # TODO: improve this
         # This is updating the graph/function outputs to use the new outputs
         for inserted_node in inserted_nodes:
             for new_output in inserted_node.outputs:
                 if (index := new_output.meta.get(_ir_utils.GRAPH_OUTPUT_META_KEY)) is not None:  # type: ignore[assignment]
                     graph_or_function.outputs[index] = new_output
 
-    graph_or_function.remove(to_delete, safe=True)
+        for d in deleted_nodes:
```

Comment:
I would rename deleted_nodes to nodes_to_remove or something like it because they are not removed yet. Non blocking

Reply:
Renamed to `matched_nodes` to match usage elsewhere